.github/workflows/publish.yml

```diff
@@ -0,0 +1,40 @@
+name: Publish Snapshot builds
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - v*
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  publish:
+    name: Snapshot build and publish
+    runs-on: macos-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+
+      - name: Set up JDK
+        uses: actions/setup-java@99b8673ff64fbf99d8d325f52d9a5bdedb8483e9 # v4.2.1
+        with:
+          java-version: 17
+          distribution: adopt
+
+      - name: Set up gradle
+        uses: gradle/actions/setup-gradle@db19848a5fa7950289d3668fb053140cf3028d43 # v3.3.2
+
+      - name: Publish to MavenCentral
+        if: github.repository == 'mihonapp/bitmap.kt'
+        run: ./gradlew publishAllPublicationsToMavenCentralRepository --no-daemon --no-parallel
```

Comment:
Should be able to remove `--no-daemon` and `--no-parallel`, the vanniktech publish plugin has fixed the bugs with those options from what I know.